### DISCUSSION TITLE
Suunnitellut toimenpiteet kipalle talvella 2024-2025

### DISCUSSION THREAD

**Discussion by ilesoft** [partio-scout/kipa#67] [General]
Olen suunnitellut tässä talvella tekeväni Kipaan useita uudistuksia:

- [x] Dokumentaation järkeistäminen
- [x] Projektin kehitystavoista sopiminen (aloitettu jo kun CONTRIBUTING.md on luotu)
- [x] Projektin testattavuuden varmistaminen
- [x] Julkaistavuuden varmistaminen
- [x] Selkeästi käyttämättömien osien poisto
- [x] Koodin ja templaattien muotoilu
- [x] Kirjastojen ja ohjelmointikielen päivittäminen (odottaa katselmointia ja korjauksia)
- [x] Lintterit käyttöön
- [ ] Tiedetyt ohjelmistovirheet (Jos kerkiää)

**Comment by pholm** [comment]
Kiitos jo nyt @ilesoft valtavasta työstä tämän projektin eteen! Lista on järkevä ja paljon onkin jo mergeilty.

Muutamia meta-ajatuksia tähän jatkoksi:

- Partiokilpailuja tullaan järjestämään jatkossakin ja Kipa on de facto ainoa "maallikolle" sopiva työkalu tulosten laskemiseen. Osaava henkilö tekee samat asiat Excelillä/Sheetsillä, mutta virheiden riski on aika suuri. Tarvetta ylläpidolle siis on.
- Tekisin pessimistisen oletuksen, että kovin aktiivista open-source yhteisöä Kipan ympärille ei ole rakentumassa. Se olisi toki hienoa ja tervetullutta, mutta samalla esimerkiksi dokumentaation, CI/CD putkien, asennuspakettien yms. rakentelussa pitäisin mielessä, että tämän pitää toimia "itsenäisesti" hyvinkin pitkään.
- Yleisesti ottaen kriittisimmäksi ongelmaksi näen Kipan asennuksen ja "alkuun pääsyn". Ihan vain vaikean asennuksen + tuen puutteen vuoksi monet saattavat valita muita vaihtoehtoja. Tämä on haitaksi myös kehitysyhteisölle, koska useimmat Kipan pariin päätyvät ovat sitä käyttämässä/käyttäneet itse tulosten laskemiseen.
- Devaamisessa alkuun pääsy on myös jonkin tason ongelma, mutta parantunut merkittävästi tässä viime viikkojen aikana :)

**Comment by mikkolindfors** [comment]
Kiitos kaikille ketkä ovat nyt talvella tehneet jotain Kipalle. Onnistuin asentamaan todella helposti Oraclen pilveen ilmaiselle virtuaalipalvelimelle. Ainoastaan vaihdoin portin 3000:3000 koska putki ulos oli siellä jo edellisten kokeilujen kautta kunnossa.
Jatkamme tästä SM-kisojen parissa ja luodaan vielä backup Kipasta. Palaan asiaan kun hölmöjä kysymyksiä ilmenee, kiitos vielä kerran helpotuksesta asennuksen kanssa.

**Comment by mikkolindfors** [comment]
Kysymyksiä ilmeni.
Asensin Kipan dockerilla eri koneille ja pilveen. aina toistui sama ongelma, useammalla yhtäaikaisella käyttäjällä sivu ei lataudu. Tästä juttelin jo tietävien tahojen kanssa ja pohdinta oli että mahdollisesti SQLlite on syyllinen eikä hanskaa useampaa käyttäjää samaan aikaan.
Ehdotukseni onkin, voisiko dockeriin vaihtaa MYSQL tai muun modernin kannan taustalle.
Docker asennus on sinällään hyvä ja helppo kohtuu maallikollekin saada käyntiin ja käyttöön. Kiitos siitä
